### PR TITLE
Improve Region Of Interest filter

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

File: Code/BasicFilters/json/RegionOfInterestImageFilter.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
       "default" : "std::vector<unsigned int>(3, 1)",
       "dim_vec" : 1,
       "itk_type" : "typename InputImageType::SizeType",
-      "briefdescriptionSet" : "",
+      "briefdescriptionSet" : "Size in pixels of the region extracted.",
```
Comment (zivy) [inline Code/BasicFilters/json/RegionOfInterestImageFilter.json]: I think it should read "extracted region." and not "region extracted."

File: Code/BasicFilters/json/RegionOfInterestImageFilter.json [hunk 2]
```diff
@@ -24,11 +24,25 @@
       "default" : "std::vector<int>(3, 0)",
       "dim_vec" : 1,
       "itk_type" : "typename InputImageType::IndexType",
-      "briefdescriptionSet" : "",
-      "detaileddescriptionSet" : "\todo the internal setting of the method need work!!!",
+      "briefdescriptionSet" : "Set the inclusive starting index of the region extracted.",
```
Comment (zivy) [inline Code/BasicFilters/json/RegionOfInterestImageFilter.json]: "extracted region."

File: Code/BasicFilters/json/RegionOfInterestImageFilter.json [hunk 3]
```diff
@@ -24,11 +24,25 @@
...
       "briefdescriptionGet" : "",
       "detaileddescriptionGet" : "",
-      "custom_itk_cast" : " typename InputImageType::RegionType itkRegion;\n  for( unsigned int i = 0; i < inImage1.GetDimension(); ++i )\n    {\n    itkRegion.SetIndex( i, m_Index[i] );\n    itkRegion.SetSize( i, m_Size[i] );\n    }\n  filter->SetRegionOfInterest( itkRegion );"
+      "custom_itk_cast" : " typename InputImageType::RegionType itkRegion(sitkSTLVectorToITK<typename InputImageType::IndexType>(m_Index), sitkSTLVectorToITK<typename InputImageType::SizeType>(m_Size));\n  filter->SetRegionOfInterest( itkRegion );"
+    }
+  ],
+  "custom_methods" : [
+    {
+      "name" : "SetRegionOfInterest",
+      "doc" : "Sets the region extracted by a single array of the starting indexes followed by the sizes.",
```
Comment (zivy) [inline Code/BasicFilters/json/RegionOfInterestImageFilter.json]: "followed by the sizes in pixels."

File: Documentation/Doxygen/Conventions.dox [hunk 4]
```diff
@@ -58,6 +58,19 @@ in row major order. The vector \f$[1,2,3,4,5,6]\f$ represents the matrix
...
 
+\section ImageRegions Image Regions as Index and Size
+
+The itk::ImageRegion is a frequently used class in ITK to define a
+sub-image area. It is defined by itk::Index and itk::Size of signed
+and unsigned integer types respecfully. In SimpleITK, the index and
+size elements are usually separated into two arguments with separate
+Set and Get methods.
+
+When needed to be specified as a single value, it is a 1 dimensional
```
Comment (zivy) [inline Documentation/Doxygen/Conventions.dox]: Possibly change to: "When specified as a single argument value, it is a 1 dimensional..."

**Review by zivy — COMMENTED** [review]
LGTM. Only a couple of minor comments.

File: Documentation/Doxygen/Conventions.dox (hunk 4 above)
Comment (blowekamp) [reply Documentation/Doxygen/Conventions.dox]: done

File: Code/BasicFilters/json/RegionOfInterestImageFilter.json (hunk 3 above)
Comment (blowekamp) [reply Code/BasicFilters/json/RegionOfInterestImageFilter.json]: done

File: Code/BasicFilters/json/RegionOfInterestImageFilter.json (hunk 2 above)
Comment (blowekamp) [reply Code/BasicFilters/json/RegionOfInterestImageFilter.json]: This comes from ITK

File: Code/BasicFilters/json/RegionOfInterestImageFilter.json (hunk 1 above)
Comment (blowekamp) [reply Code/BasicFilters/json/RegionOfInterestImageFilter.json]: This comes from ITK

**Pull request merged**